.github/dependabot.yml

```diff
@@ -0,0 +1,21 @@
+---
+version: 2
+updates:
+  # Enable version updates for nuget
+  - package-ecosystem: "nuget"
+    # Look for the dependency files in the `root` directory
+    directory: "/"
+    # Check for updates once a week
+    schedule:
+      interval: "weekly"
+    reviewers:
```

Comment:
```suggestion
    allowed_updates:
      - match:
          dependency_name: "Elastic.Apm.NetCoreAll"
    reviewers:
```

Reply:
This particular dependency is already automated as part of the apm-agent-dotnet release pipeline.
The idea is to automate the bump for the other dependencies if it makes any sense

Reply:
Oh, I see. 

> The idea is to automate the bump for the other dependencies if it makes any sense

We have 2 other dependencies here:
- `AutoMapper`  - we use very minimal functionality from it and it really doesn't matter which version we use, all it does is that it mapes database types to DTO types 
- `Microsoft.EntityFrameworkCore.Sqlite` that's the ORM library to talk to the database. We use a fairly old version, I'm not sure what'd happen if we'd bump it to the newest version, likely it'd work out of the box.

I don't have strong opinion - I think it's fine to update those as well, but I don't see it as high prio.

> This particular dependency is already automated as part of the apm-agent-dotnet release pipeline.

But that doesn't update the file here in the repo, right? The latest release is `1.9.0` but we still have `1.8.1` as a dependency here - I think best would be to just update the dependency here in the repo on the file level as well. Either way if it's done somewhere else then I think we should invert the `allow_updates` config (to not do it twice) or make sure that we update it only either with dependabot or as part of the release pipeline.

Reply:
> But that doesn't update the file here in the repo, right? The latest release is 1.9.0 but we still have 1.8.1 as a dependency here - I think best would be to just update the dependency here in the repo on the file level as well. Either way if it's done somewhere else then I think we should invert the allow_updates config (to not do it twice) or make sure that we update it only either with dependabot or as part of the release pipeline.

We can exclude it from dependabot sure, though this process runs on a weekly basis. There is a low chance there will be a clash, though, in case there was, the dependabot won't change anything but raise a PR.

If you prefer we can add the exclusion

Reply:
> There is a low chance there will be a clash, though, in case there was, the dependabot won't change anything but raise a PR.

Since this runs weekly and even if there is a clash I think it's fine - now that you say I personally don't think an exclude is needed.

So - how do we proceed? After @v1v 's comments I think we don't need these `allowed_updates` part - we could just let open PRs for newer versions for every dependency - so merge this PR as it is.

Let's see what @russcam says.